src/Core/Domain/Cart/Command/UpdateProductQuantityInCartCommand.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Cart\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Cart\ValueObject\CartId;
+use PrestaShop\PrestaShop\Core\Domain\Cart\ValueObject\QuantityAction;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+
+/**
+ * Updates product quantity in cart
+ */
+class UpdateProductQuantityInCartCommand
+{
+    /**
+     * @var CartId
+     */
+    private $cartId;
+
+    /**
+     * @var ProductId
+     */
+    private $productId;
+
+    /**
+     * @var int
+     */
+    private $quantity;
+
+    /**
+     * @var QuantityAction
+     */
+    private $action;
+
+    /**
+     * @var int|null
+     */
+    private $combinationId;
+
+    /**
+     * @var int|null
+     */
+    private $customizationId;
+
+    /**
+     * @param int $cartId
+     * @param int $productId
+     * @param int $quantity
+     * @param string $action
+     * @param int|null $combinationId
+     * @param int|null $customizationId
+     */
+    public function __construct(
+        $cartId,
+        $productId,
+        $quantity,
+        $action,
+        $combinationId = null,
+        $customizationId = null
+    ) {
+        // @todo: add integirty checks for $quantity, $combinationId, $customizationId
+
+        $this->cartId = new CartId($cartId);
+        $this->productId = new ProductId($productId);
+        $this->quantity = $quantity;
```

Comment:
Should there be `Quantity` VO? :thinking: For me it kind of makes sense, because not any `int` can be quantity. But maybe it's redundant and simple integrity check during command creation is sufficient?

Reply:
Let's stick with `int` if we dont see the need for a VO ey